Polyclinic/Polyclinic.Api/Controllers/AnalyticsController.cs

```diff
@@ -0,0 +1,129 @@
+﻿using Microsoft.AspNetCore.Mvc;
+using Polyclinic.Contracts;
+using Polyclinic.Contracts.Appointments;
+using Polyclinic.Contracts.Doctors;
+using Polyclinic.Contracts.Patients;
+
+namespace Polyclinic.Api.Controllers;
+
+/// <summary>
+/// Controller for handling analytical data and reports
+/// </summary>
+public class AnalyticsController(IAnalyticsService service, ILogger<AnalyticsController> logger) : Controller
+{
+    /// <summary>
+    /// Retrieves doctors with experience at least specified number of years
+    /// </summary>
+    /// <param name="minYears">Minimum years of experience required</param>
+    /// <returns>List of doctors meeting the experience criteria</returns>
+    [HttpGet("doctors/experienced/{minYears}")]
+    [ProducesResponseType(200)]
+    [ProducesResponseType(204)]
+    [ProducesResponseType(500)]
+    public ActionResult<List<DoctorDto>> GetDoctorsWithExperienceAtLeast(int minYears)
```

Comment:
Не мешало бы добавить валидацию кол-ва лет. Это можно сделать через Range валидационный атрибут.
```suggestion
    public ActionResult<List<DoctorDto>> GetDoctorsWithExperienceAtLeast([Range(0,100)] int minYears)
```
Аналогично можно докинуть в других местах контроллера.